### DISCUSSION TITLE
Docs clarification

### DISCUSSION THREAD

**Discussion by sztomi** [compiler-explorer/compiler-explorer#5017] [Q&A] [answered]
[Adding a new compiler](https://github.com/compiler-explorer/compiler-explorer/blob/main/docs/AddingACompiler.md) says the following about config file hierarchies: 

<blockquote>
By design, this does not however work for groups (Nor any other nested property). That is, if in etc/config/c++.defaults.properties you define the intelAsm property as:

```
versionFlag=--version
compilers=&clang
group.clang.intelAsm=-mllvm -x86-asm-syntax=intel
group.clang.groupName=Clang
...
```
but etc/config/c++.amazon.properties only has:
```
compilers=&clang
group.clang.groupName=Clang
...
```
</blockquote>

The way I understand it is that the properties will be inherited from defaults, but not in the case of groups or other nested properties (by design). But the example seems to contradict what is stated: `group.clang.groupName` is also inherited. Or is the group special somehow, or is the example wrong? Or am I misunderstanding what's said?

**Comment by partouf** [comment, marked as answer]
All things get overridden if you name them the same, in this case `compilers` and `group.clang`, but `versionFlag` is not overridden.

**Reply by sztomi** [reply]
Thanks, I misunderstood it after all, totally makes sense.
